include/openthread/dataset.h

```diff
@@ -605,4 +606,27 @@ otError otDatasetUpdateTlvs(const otOperationalDataset *aDataset, otOperationalD
 } // extern "C"
 #endif
 
+#ifdef __cplusplus
+/**
+ * Defines equality for two Timestamps.
+ */
+inline bool operator==(const otTimestamp &aLhs, const otTimestamp &aRhs)
+{
+    return (aLhs.mSeconds == aRhs.mSeconds) && (aLhs.mTicks == aRhs.mTicks) &&
+           (aLhs.mAuthoritative == aRhs.mAuthoritative);
+}
+
+/**
+ * Defines equality for two Operational Datasets.
+ *
+ * Equality is intentionally scoped to the Active Timestamp and Extended PAN ID
+ * which are minimal set of fields needed to identify a Thread network.
+ */
+inline bool operator==(const otOperationalDataset &aLhs, const otOperationalDataset &aRhs)
+{
+    return (aLhs.mActiveTimestamp == aRhs.mActiveTimestamp) &&
+           (memcmp(aLhs.mExtendedPanId.m8, aRhs.mExtendedPanId.m8, OT_EXT_PAN_ID_SIZE) == 0);
```

Comment:
This implementation does not account for the fact that all properties/components (such as `mActiveTimestamp` or `mExtendedPanId`) in an `otOperationalDataset` struct are optional. 

Please check the `otOperationalDatasetComponents` (e.g., you need to check `mIsActiveTimestampPresent` before accessing `mActiveTimestamp`).